integrations/chroma/src/haystack_integrations/document_stores/chroma/document_store.py

```diff
@@ -81,13 +85,21 @@ def __init__(
             self._chroma_client = chromadb.PersistentClient(path=persist_path)
 
         embedding_func = get_embedding_function(embedding_function, **embedding_function_params)
-        metadata = {"hnsw:space": distance_function}
+
+        if metadata is None:
+            metadata = {}
+        if "hnsw:space" not in metadata:
+            metadata["hnsw:space"] = distance_function
 
         if collection_name in [c.name for c in self._chroma_client.list_collections()]:
             self._collection = self._chroma_client.get_collection(collection_name, embedding_function=embedding_func)
 
             if distance_function != self._collection.metadata["hnsw:space"]:
                 logger.warning("Collection already exists. The `distance_function` parameter will be ignored.")
+
+            for key in metadata:
+                if metadata[key] != self._collection.metadata[key]:
+                    logger.warning(f"Collection already exists. The `metadata[{key}]` parameter will be ignored.")
```

Comment:
```suggestion
            if metadata != self._collection.metadata:
                logger.warning("Collection already exists. The `distance_function` and `metadata` parameters will be ignored.")
```

What about simplifying this check?

Reply:
I expected chromadb would implicitly populate its `self._collection.metadata` dictionary with default keys and values. I iterated over `metadata` keys to avoid getting a warning if, for example, the user didn’t provide any dictionary and `metadata` is `{}` and `self._collection.metadata` has default keys and values set by chromadb.

I just checked though, it doesn’t seem to be the case. I suppose this will work.